flink-libraries/flink-table/src/main/scala/org/apache/flink/api/table/table.scala

```diff
@@ -590,7 +592,7 @@ class Table(
       .map(field => FlinkTypeFactory.toTypeInfo(field.getType)).toArray
 
     // configure the table sink
-    val configuredSink = sink.configure(fieldNames, fieldTypes)
+    val configuredSink = sink.asInstanceOf[TableSinkBase[T]].configure(fieldNames, fieldTypes)
```

Comment:
It seems that KafkaTableSink doesn't extends TableSinkBase, so that does it work if the input sink is a KafkaTableSink ?